Packs/Axonius/Integrations/Axonius/Axonius.py

```diff
@@ -0,0 +1,213 @@
+import traceback
+from typing import Any, List, Optional, Union
+
+import demistomock as demisto
+from axonius_api_client.api.assets.devices import Devices
+from axonius_api_client.api.assets.users import Users
+from axonius_api_client.connect import Connect
+from axonius_api_client.tools import dt_parse, strip_left
+from CommonServerPython import *
+from CommonServerUserPython import *
+
+MAX_ROWS: int = 50
+"""Maximum number of assets to allow user to fetch."""
+SKIPS: List[str] = ["specific_data.data.image"]
+"""Fields to remove from each asset if found."""
+FIELDS_TIME: List[str] = ["seen", "fetch", "time", "date"]
+"""Fields to try and convert to date time if they have these words in them."""
+
+
+def get_int_arg(
+    key: str, required: Optional[bool] = False, default: Optional[int] = None,
+) -> int:
+    """Get a key from a command arg and convert it into an int."""
+    args: dict = demisto.args()
+    value: int = args.get(key, default)
+
+    if value is None and required:
+        raise ValueError(f"No value supplied for argument {key!r}")
+
+    try:
+        return int(value)
+    except Exception:
+        raise ValueError(
+            f"Supplied value {value!r} for argument {key!r} is not an integer."
+        )
+
+
+def get_csv_arg(
+    key: str, required: Optional[bool] = False, default: Optional[str] = "",
+) -> List[str]:
+    """Get string values from CSV."""
+    args: dict = demisto.args()
+    value: str = args.get(key, default)
+
+    if not value and required:
+        raise ValueError(f"No value supplied for argument {key!r}")
+
+    return [x.strip() for x in value.split(",") if x.strip()]
+
+
+def test_module(client: Connect) -> str:
+    """Tests Axonius API Client connectivity."""
+    client.start()
+    return "ok"
+
+
+def parse_kv(key: str, value: Any) -> Any:
+    """Parse time stamp into required format."""
+    for word in FIELDS_TIME:
+        if word in key:
+            try:
+                return dt_parse(value).isoformat()
+            except Exception:
+                return value
+    return value
+
+
+def parse_key(key: str) -> str:
+    """Parse fields into required format."""
+    if key.startswith("specific_data.data."):
+        key = strip_left(obj=key, fix="specific_data.data.")
+        key = f"aggregated_{key}"
+    if key.startswith("adapters_data."):
+        key = strip_left(obj=key, fix="adapters_data.")
+    key = key.replace(".", "_")
+    return key
+
+
+def parse_asset(asset: dict) -> dict:
+    """Initiate field format correction on assets."""
+    return {
+        parse_key(key=k): parse_kv(key=k, value=v)
+        for k, v in asset.items()
+        if k not in SKIPS
+    }
+
+
+def check_asset_count(api_obj: Union[Users, Devices], assets: List[dict], src: str):
+    """Raise exception for no assets found."""
+    if not assets:
+        api_name = api_obj.__class__.__name__
+        raise Exception(f"No {api_name} assets returned from {src}.")
```

Comment:
don't raise an Excpetion here, just return the expected human readable to the command and build a CommandResults object with no outputs and the human readable returned.

Reply:
@guyfreund When working with @fvigo we were told to raise an exception when no assets were found. Wouldn't the human readable output for this just be nothing or an empty list? We can just return an empty list if you would rather we didn't raise an exception.

Reply:
@Phillips-Chris no need to raise an exception. the human readable should be `No assets were found.`

Reply:
@guyfreund Got it, updated the code appropriately.